app/notification/schedular/context_aware_executor.py

```diff
@@ -0,0 +1,24 @@
+from concurrent.futures import ThreadPoolExecutor
+from contextvars import copy_context
+
+
+class ContextAwareExecutor:
+
+    def __init__(self, max_workers, thread_name_prefix, flask_app):
+        self.executor = ThreadPoolExecutor(max_workers=max_workers, thread_name_prefix=thread_name_prefix)
+        self.flask_app = flask_app
+
+    def queue_size(self):
+        return self.executor._work_queue.qsize()
+
+    def submit(self, fn, *args, **kwargs):
+        ctx = copy_context()
+        future = self.executor.submit(ctx.run, self.wrap_function(fn), *args, **kwargs)
+        return future
+
+    def wrap_function(self, fn):
```

Comment:
Do we actually need to run each thread under a flask context? It looks like each task only calls `Notification.email_recipient`

Reply:
Without flask context I don't have the logger and flask resources mainly the loggers